modules/gdscript/gdscript_parser.cpp

```diff
@@ -1704,17 +1657,15 @@ void GDScriptParser::parse_function_signature(FunctionNode *p_function, SuiteNod
 	}
 #endif // TOOLS_ENABLED
 
-	if (p_function->is_abstract) {
-		end_statement("abstract function declaration");
-	} else {
-		// TODO: Improve token consumption so it synchronizes to a statement boundary. This way we can get into the function body with unrecognized tokens.
-		consume(GDScriptTokenizer::Token::COLON, vformat(R"(Expected ":" after %s declaration.)", p_type));
+	if (p_type == "lambda") {
+		return consume(GDScriptTokenizer::Token::COLON, vformat(R"(Expected ":" and body after lambda function header.)"));
 	}
+	// The colon may not be present in the case of abstract functions.
+	return match(GDScriptTokenizer::Token::COLON);
```

Comment:
```suggestion
	// TODO: Improve token consumption so it synchronizes to a statement boundary. This way we can get into the function body with unrecognized tokens.
	if (p_type == "lambda") {
		return consume(GDScriptTokenizer::Token::COLON, R"(Expected ":" after lambda declaration.)");
	}
	// The colon may not be present in the case of abstract functions.
	return match(GDScriptTokenizer::Token::COLON);
```

I'm not sure if the comment is still relevant, but it's probably best left unchanged.